### PR TITLE
chore: remove personal branding if in an org and use org branding

### LINKED ISSUE #14090: [CAL-3398] remove personal branding if you are in an org and use org branding
we should get rid of the personal "apperance" tab in the personal profile section if you are part of an org

CleanShot 2024-03-14 at 13 19 25@2x

and instead use all of the branding settings from the org:

CleanShot 2024-03-14 at 13 20 01@2x

we should also merge "Appearance" into "Profile" 

CleanShot 2024-03-14 at 13 23 22@2x

<sub>[CAL-3398](https://linear.app/calcom/issue/CAL-3398/remove-personal-branding-if-you-are-in-an-org-and-use-org-branding)</sub>

### PR DESCRIPTION
**Pull request by anikdhabal**
## What does this PR do?

Fixes #14090 
Screenshot 2024-05-04 002125
Screenshot 2024-05-04 002100

### CODE REVIEW

File: apps/web/pages/settings/my-account/appearance.tsx [hunk 1]
```diff
@@ -430,7 +432,11 @@ const AppearanceView = ({
 const AppearanceViewWrapper = () => {
   const { data: user, isPending } = trpc.viewer.me.useQuery();
   const { isPending: isTeamPlanStatusLoading, hasPaidPlan } = useHasPaidPlan();
-
+  const session = useSession();
+  const router = useRouter();
+  if (!!session.data?.user?.org?.id) {
+    router.replace("/settings/organizations/profile");
+  }
```
Comment (Udit-takkar) [inline apps/web/pages/settings/my-account/appearance.tsx:439]: I think we should just hide the tab instead of doing this

File: packages/features/ee/organizations/pages/settings/appearance.tsx [hunk 2]
```diff
@@ -79,134 +75,79 @@ const OrgAppearanceView = ({
   };
 
   return (
-    <LicenseRequired>
-      <Meta
-        title={t("appearance")}
-        description={t("appearance_org_description")}
-        borderInShellHeader={false}
-      />
```
Comment (Udit-takkar) [inline packages/features/ee/organizations/pages/settings/appearance.tsx:87]: Why did you remove LicenseRequired and Meta ?

Comment (Udit-takkar) [reply packages/features/ee/organizations/pages/settings/appearance.tsx:87]: and only

File: packages/features/ee/organizations/pages/settings/appearance.tsx [hunk 3]
```diff
@@ -27,7 +24,6 @@ type BrandColorsFormValues = {
 
 const OrgAppearanceView = ({
   currentOrg,
-  isAdminOrOwner,
```
Comment (Udit-takkar) [inline packages/features/ee/organizations/pages/settings/appearance.tsx:30]: Only admin or owner should be able to change organization's appearance settings.

File: packages/features/settings/layouts/SettingsLayout.tsx [hunk 4]
```diff
@@ -185,6 +181,14 @@ const useTabs = () => {
...
     });
 
+    if (!!session.data?.user?.org?.id) {
+      processedTabs.forEach((tab) => {
+        if (tab.href === "/settings/my-account") {
+          tab.children = tab?.children?.filter((child) => child.href !== "/settings/my-account/appearance");
+        }
+      });
+    }
+
```
Comment (Udit-takkar) [inline packages/features/settings/layouts/SettingsLayout.tsx:191]: No need to do the iteration again. you can filter the children inside     

const processedTabs = tabs.map((tab) => {
      if (tab.href === "/settings/my-account") {

File: packages/features/ee/organizations/pages/settings/appearance.tsx (hunk 2 above)
Comment (anikdhabal) [reply packages/features/ee/organizations/pages/settings/appearance.tsx:87]: We used the component in the profile section, where LicenseRequired and Meta were already used as parents.

File: packages/features/ee/organizations/pages/settings/appearance.tsx (hunk 3 above)
Comment (anikdhabal) [reply packages/features/ee/organizations/pages/settings/appearance.tsx:30]: We handle it in the profile section. Do we need to check each one individually?

File: packages/features/settings/layouts/SettingsLayout.tsx (hunk 4 above)
Comment (anikdhabal) [reply packages/features/settings/layouts/SettingsLayout.tsx:191]: Ahh, sure.

File: apps/web/pages/settings/my-account/appearance.tsx (hunk 1 above)
Comment (anikdhabal) [reply apps/web/pages/settings/my-account/appearance.tsx:439]: Hey Udit, I already hide this tab in the settings layout. We added it because if a user tries to access the page using the link, then we redirect them to the profile page.

**Comment by anikdhabal** [comment]
Hey @Udit-takkar any update on this one?

**Review by Udit-takkar — DISMISSED** [review]
LGTM. we might have to remove /org/settings/appearance page later

**Review by CarinaWolli — DISMISSED** [review]
e2e test in `upload-avatar.e2e.ts` is failing

**Comment by anikdhabal** [comment]
> e2e test in `upload-avatar.e2e.ts` is failing

Thanks @CarinaWolli for the review. I will fix that shortly.

**Comment by CarinaWolli** [comment]
`it can upload a organization logo image` in `upload-avatar.e2e.ts` is still failing for me 🤔

**Review by Udit-takkar — COMMENTED** [review]
I have fixed the E2E test but there is also an open issue to change the UI of banner uploader https://github.com/calcom/cal.com/issues/15456

**Comment by anikdhabal** [comment]
> I have fixed the E2E test but there is also an open issue to change the UI of banner uploader #15456

Hey Udit, thanks for it. I've gone through his PR, and he completely created the OrgBanner component and imported and used it in the profile. So after this merge, it won't be tough to update the code. He just needs to import and use it in different components. And obviously, these two issues are different.

**Review by Udit-takkar — APPROVED** [review]
